js/private/yq.bzl

```diff
@@ -0,0 +1,24 @@
+"Utilities for using the yq toolchain in a repository rule"
+
+load("@aspect_bazel_lib//lib:repo_utils.bzl", "is_windows_os")
+
+def yq_bin(rctx, yq_repository):
+    """Locate the path to a yq binary.
+
+    Args:
+        rctx: the repository context
+        yq_repository: the name of the yq toolchain repository
+
+    Returns:
+        Path to the yq binary
+    """
+
+    # Parse the resolved host platform from yq host repo //:index.bzl
+    content = rctx.read(rctx.path(Label("@%s_host//:index.bzl" % yq_repository)))
+    search_str = "host_platform=\""
```

Comment:
gross! we shouldn't do this - instead the `yq_register_toolchains(name = "yq")` should produce a repository `@yq` which is for the host platform

Reply:
I think this is a workaround for an issue @gregmagolan found when we tried making a target pointing to the resolved executable.

Reply:
yeah probably that `rctx.path` doesn't follow symlinks - but we figured out a way around it in rules_python without ever trying to predict the repo name

Reply:
Are you referring to https://github.com/bazelbuild/rules_python/blob/ae7a2677b3003b13d45bc9bfc25f1425bed5b407/python/pip_install/pip_repository.bzl#L45?

Reply:
Instead of a repository with aliases we could just create a `@yq_host` or `@yq` repository that is a duplicate of `@yq_<platform>`. This is a bit of duplication on disk _but_ it makes the path to the yq for the host platform simple.